src/facade/dragonfly_connection.cc

```diff
@@ -784,7 +795,18 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
       }
     };
 
-    builder->SetBatchMode(dispatch_q_.size() > 1);
+    // We really want to have batching in the builder if possible. This is especially
+    // critical in situations where Nagle's algorithm can introduce unwanted high
+    // latencies. However we can only batch if we're sure that there are more commands
+    // on the way. To know if there are, we sometimes yield before executing the last
+    // command in the queue and let the producer fiber push more commands if it wants
+    // to.
+    // As a small optimization, if the queue was never larger than 1, skip the Yield() call.
+    if (dispatch_q_.size() == 1 && queue_was_larger_than_1) {
+      ThisFiber::Yield();
+    }
```

Comment:
With `memtier_benchmark --pipeline=2` you'll yield a lot 🤔 But its still more effective, right?

Reply:
It would also be more efficient to check yield only if we didn't yield on the previous iteration (i.e. the transaction was run with inline scheduling)

Reply:
yes, we need helio machinery for this. Moreover, iouring api allows learning if there is more data pending in the network buffer pending after the last `Recv` call (IORING_CQE_F_SOCK_NONEMPTY from 5.19).  I do not think I implemented support for this.

Reply:
It does trigger quite a few more Yields, I'm not sure what's the performance impact

Reply:
Updated to use the new helio epoch interface. Looked at the logs with memtier_benchmark and the JuiceFS benchmark and it looks quite good, probably 1 yield every 10 commands or something.